docs_nnx/guides/filters_guide.ipynb

```diff
@@ -5,12 +5,18 @@
    "id": "95b08e64",
    "metadata": {},
    "source": [
-    "# Using Filters\n",
+    "# Using `Filter`s\n",
     "\n",
-    "> **Attention**: This page relates to the new Flax NNX API.\n",
+    "Flax NNX uses [`Filter`s](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/filterlib.html) extensively as a way to create [`nnx.State`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/state.html#flax.nnx.State) groups in APIs, such as [`nnx.split`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/graph.html#flax.nnx.split), [`nnx.state()`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/graph.html#flax.nnx.state), and many of the [Flax NNX transformations (transforms)](https://flax.readthedocs.io/en/latest/guides/jax_and_nnx_transforms.html).\n",
     "\n",
-    "Filters are used extensively in Flax NNX as a way to create `State` groups in APIs\n",
-    "such as `nnx.split`, `nnx.state`, and many of the Flax NNX transforms. For example:"
+    "In this guide you will learn:\n",
+    "\n",
+    "* What is a `Filter`?\n",
```

Comment:
Questions without answers make subsequent questions hard to understand. What about:

In this guide you will learn to:
* Use `Filter` to group Flax variables and states into subgroups;
* Relations between types such as ... and `Filter`s;
* Express your `Filter`s flexibly with `nnx.filterlib.Filter` language.

(I don't think the DSL terminology is important here as we can keep the overview more concise and understandable)

Reply:
SGTM!

```
* Use [`Filter`s](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/filterlib.html) to group Flax NNX variables and states into subgroups;
* Understand relationships between types, such as [`nnx.Param`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/variables.html#flax.nnx.Param) or [`nnx.BatchStat`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/variables.html#flax.nnx.BatchStat), and [`Filter`s](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/filterlib.html)?
* Express your `Filter`s flexibly with [`nnx.filterlib.Filter`](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/filterlib.html) language.
* ```